usaspending_api/bulk_download/v2/views.py

```diff
@@ -307,6 +307,8 @@ def post(self, request):
                           'end_date': '{}-09-30'.format(fiscal_year)}
             bulk_download_filters.update(date_range)
         if agency_id:
+            if agency_id == 'all':
+                agency_id = None
```

Comment:
I don't fully understand the decision to set agency_id to None if it is "all" and add it to the bulk_download_filters, but if the agency_id is originally None, it isn't added to the bulk_download_filters. Would it be better to do something like this if statement?
`if agency_id and agency_id != 'all':`

Reply:
Yeah that's the problem of the wording here. There's a difference between "get the monthly files that represent all the agencies for a fiscal year or a fiscal type" (`all_contracts_2017`, `all_assistance_2017`),  vs "get all agency files" (`dod_contracts_2017`, `doe_contracts_2017`,`all_contracts_2017`,`all_assistance_2017`). So to handle this difference, if `agency='all'`, it will only include the "all" files whereas removing it will include the all files as well as the other agency files.